File: sgf-poker/sgf-poker-ios/Core/Repositories/PlayerRepository.swift
// Core/Repositories/PlayerRepository.swift
import Foundation

/// Contract for any players data source.
protocol PlayerRepository: AnyObject {
    /// Returns all stored players, sorted by name ascending.
    func fetchAll() throws -> [Player]

    /// Persists a new player. Throws if a player with the same name exists.
    func save(_ player: Player) throws

    /// Removes a player by its identifier.
    func delete(id: UUID) throws
}

File: sgf-poker-android/app/src/main/java/com/sgf/poker/data/repository/PlayerRepository.java
package com.sgf.poker.data.repository;

import com.sgf.poker.domain.model.Player;
import java.util.List;

/** Contract for any players data source. */
public interface PlayerRepository {
    /** Returns all players sorted alphabetically by name. */
    List<Player> fetchAll();

    /** Persists a player. Throws PlayerException.duplicateName if name already exists. */
    void save(Player player);

    /** Removes a player by id. Throws PlayerException.notFound if missing. */
    void delete(String id);
}
